### PR TITLE
Feat/create app modal

### PR DESCRIPTION
**Pull request by NNaumenko83**
Hello everyone,

I have prepared the AppModal. Please take a moment to review and provide your feedback, or if everything is all right, kindly approve.

### CODE REVIEW

File: src/shared/AppModal.jsx/AppModal.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * AppModal component for rendering a AppModal window.
+ *
+ * @component
+ * @param {function} handleClose - The callback function to handle modal close.
+ * @param {boolean} open - A boolean indicating whether the modal is open or closed.
+ * @param {React.ReactNode} children - The content to be displayed within the modal.
+ * @returns {JSX.Element}
+ */
+
+export const AppModal = ({ handleClose, children, open }) => {
```
Comment (Zaimak-Andrii) [inline src/shared/AppModal.jsx/AppModal.jsx:18]: Maybe I changed handleClose to onClose. It's only a proposition. :)

File: src/shared/AppModal.jsx/AppModal.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { AiOutlineCloseCircle } from 'react-icons/ai';
```
Comment (Zaimak-Andrii) [inline src/shared/AppModal.jsx/AppModal.jsx:1]: AppModal.jsx folder name:) Misprint?

File: src/shared/AppModal.jsx/AppModal.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import Modal from '@mui/material/Modal';
+
+import { AppModalProps } from './AppModal.props';
+import { BoxStyled, IconButtonStyled } from './Modal.styled';
```
Comment (Zaimak-Andrii) [inline src/shared/AppModal.jsx/AppModal.jsx:6]: Incorrect import. Maybe 'AppModal.styled'.

File: src/shared/AppModal.jsx/AppModal.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * @param {boolean} open - A boolean indicating whether the modal is open or closed.
+ * @param {React.ReactNode} children - The content to be displayed within the modal.
+ * @returns {JSX.Element}
+ */
+
+export const AppModal = ({ handleClose, children, open }) => {
+  return (
+    <Modal keepMounted open={open} onClose={handleClose}>
+      <BoxStyled>
+        <IconButtonStyled
```
Comment (Zaimak-Andrii) [inline src/shared/AppModal.jsx/AppModal.jsx:22]: Screenshot 2023-11-13 at 15 33 42
Icon can be so large?

**Review by Zaimak-Andrii — CHANGES_REQUESTED** [review]

**Review by 4m4rel — APPROVED** [review]

File: src/shared/AppModal.jsx/AppModal.jsx (hunk 2 above)
Comment (NNaumenko83) [reply src/shared/AppModal.jsx/AppModal.jsx:1]: Experiencing the consequences of both working and coding simultaneously))

File: src/shared/AppModal.jsx/AppModal.jsx (hunk 4 above)
Comment (NNaumenko83) [reply src/shared/AppModal.jsx/AppModal.jsx:22]: Ok. I will change it. In the future, we can always change this button depending on our final needs. But for the reviewsForm in modal, my button looks normal.

**Review by Zaimak-Andrii — APPROVED** [review]

**Pull request merged**